catalog/core/catalog-core-impl/pubsub/src/main/java/ddf/catalog/pubsub/internal/SubscriptionFilterVisitor.java

```diff
@@ -379,6 +377,19 @@ public Object visit(During filter, Object data) {
         return returnPredicate;
     }
 
+    private DateType getDateType(String attr) {
+        if (StringUtils.isEmpty(attr)) {
+            return null;
```

Comment:
❗️ [`valueOf()`](https://docs.oracle.com/javase/7/docs/api/java/lang/Enum.html#valueOf(java.lang.Class,%20java.lang.String)) throws a `NullPointerException` if the argument is `null` and `IllegalArgumentException` if it's invalid. We may want to be consistent here and do the same to avoid problems.